flask_securest/authentication_providers/password.py

```diff
@@ -36,29 +39,45 @@ class PasswordAuthenticator(AbstractAuthenticationProvider):
 
     def __init__(self, password_hash=DEFAULT_PASSWORD_HASH):
         self.crypt_ctx = _get_crypt_context(password_hash)
+        self.request_user_id = None
+        self.request_password = None
 
-    def authenticate(self, auth_info, userstore):
-        if not auth_info.user_id or not auth_info.password:
-            raise Exception('username or password not found on request')
+    def _retrieve_request_credentials(self):
+        auth_header = request.headers.get(AUTH_HEADER_NAME)
+        if not auth_header:
+            raise RuntimeError('Request authentication header "{0}" is empty '
+                               'or missing'.format(AUTH_HEADER_NAME))
 
-        user_id = auth_info.user_id
-        user = userstore.get_user(user_id)
+        auth_header = auth_header.replace(BASIC_AUTH_PREFIX + ' ', '', 1)
+        try:
+            from itsdangerous import base64_decode
+            api_key = base64_decode(auth_header)
+            # TODO parse better, with checks and all, this is shaky
+        except TypeError:
+            pass
+        else:
+            api_key_parts = api_key.split(':')
+            self.request_user_id = api_key_parts[0]
+            self.request_password = api_key_parts[1]
+            if not self.request_user_id or not self.request_password:
+                raise RuntimeError('username or password not found on request')
 
-        if not user:
-            raise Exception('failed to authenticate user "{0}", user not found'
-                            .format(user_id))
-        if not user.password:
-            raise Exception('failed to authenticate user "{0}", password not '
-                            'found on user object'.format(user_id))
+    def authenticate(self, userstore):
+            self._retrieve_request_credentials()
```

Comment:
i think it would be better if `_retrieve_request_credentials` would simply return the `username` and `password` instead of setting them on `self`. it will make the code easier to read as there is no real need for storing them on `self`